addons/hr_timesheet/models/project.py

```diff
@@ -291,8 +291,10 @@ def _compute_analytic_account_active(self):
 
     @api.depends('timesheet_ids.unit_amount')
     def _compute_effective_hours(self):
+        timesheet_read_group = self.env['account.analytic.line'].read_group([('task_id', 'in', self.ids)], ['unit_amount', 'task_id'], ['task_id'])
+        timesheets_per_task = {res['task_id'][0]: res['unit_amount'] for res in timesheet_read_group}
         for task in self:
-            task.effective_hours = round(sum(task.timesheet_ids.mapped('unit_amount')), 2)
+            task.effective_hours = round(timesheets_per_task.get(task.id, 0.0), 2)
```

Comment:
Unfortunately we cannot just do the `read_group` because when we are in the form view of a task and we change the unit_amount of a timesheet or add a new timesheet, we want to have the correct effective hours before saving the changes.
I propose to compare `self` with `self._origin` to see if we have a NewId in the self (there is probably another solution). If it is the case then we will do.
```py
        for task in self:
            task.effective_hours = round(sum(task.timesheet_ids.mapped('unit_amount')), 2)
```
Otherwise we will do the `read_group`.
In resume, we will have this: 
```py
        if self != self._origin:
            for task in self:
                task.effective_hours = round(sum(task.timesheet_ids.mapped('unit_amount')))
            return
        timesheet_read_group = self.env['account.analytic.line'].read_group([('task_id', 'in', self.ids)], ['unit_amount', 'task_id'], ['task_id'])
        timesheets_per_task = {res['task_id'][0]: res['unit_amount'] for res in timesheet_read_group}
        for task in self:
            task.effective_hours = round(timesheets_per_task.get(task.id, 0.0), 2)

```

Reply:
can you use `fields.Float.round` to round ?